src/cli/cli.cpp

```diff
@@ -3835,12 +3835,26 @@ void Interpreter::ProcessUnsecurePort(uint8_t aArgsLength, char *aArgs[])
 
 void Interpreter::ProcessVersion(uint8_t aArgsLength, char *aArgs[])
 {
-    OT_UNUSED_VARIABLE(aArgsLength);
-    OT_UNUSED_VARIABLE(aArgs);
+    otError error = OT_ERROR_NONE;
 
-    const char *version = otGetVersionString();
-    OutputLine("%s", static_cast<const char *>(version));
-    OutputResult(OT_ERROR_NONE);
+    if (aArgsLength == 0)
+    {
+        const char *version = otGetVersionString();
+        OutputLine("%s", static_cast<const char *>(version));
+        ExitNow();
+    }
+
+    if (strcmp(aArgs[0], "api") == 0)
+    {
+        OutputLine("%d", OPENTHREAD_API_VERSION);
+    }
+    else
+    {
+        ExitNow(error = OT_ERROR_INVALID_ARGS);
```

Comment:
Should the error be `OT_ERROR_INVALID_COMMAND`?

Reply:
I think both are OK. The difference is to consider this as a subcommand or as an argument. I've seen some commands using mixed ideas, e.g. `mac` outputs invalid args but `mac something_invalid` outputs invalid command.

Reply:
My understanding is if the command or sub-command accepts parameters and the parameters are invalid, the error code would be `OT_ERROR_INVALID_ARGS`, otherwise `OT_ERROR_INVALID_COMMAND` should be generally used to indicate the command or sub-command is not supported.

The command `version`,  it doesn't have any parameters; it could support sub-command, any following word after `version` should be interpreted as a sub-command, and check whether or not it's a valid sub-command in the valid sub-command set (here only one `api` element).

I think we'd better to define a general rule to follow, other than the ambiguity in existing usages in OT like the example of `mac`that @Moandor-y  mentioned.

@wgtdkp your thoughts? is `mac` intentionally designed as is now?

Reply:
I think you are right. In the documentations something marked with brackets (like `<port>`) should be considered as an argument, otherwise it would have a separate entry in the command list and should be considered as a sub-command. In this case `version api` is a sub-command.

Reply:
@librasungirl Agree that we should follow a common rule of returning INVALID_COMMAND or INVALID_ARGS. The `mac` command returns `OT_INVALID_ARGS` is not intentional but an escaped case when introduced `OT_INVALID_COMMAND` (Will fix it soon).

@Moandor-y proposed a nice way to distinguish between an argument and sub-command:

> In the documentations something marked with brackets (like <port>) should be considered as an argument, otherwise it would have a separate entry in the command list and should be considered as a sub-command.

So, in the documentation:
1. Arguments should always be surrounded by `<>`.
2. Commands and sub-commands should always use non-escaped plain text in lowercase.

In the CLI interpreter:
1. Always return `OT_INVALID_COMMAND` for commands & sub-commands.

Reply:
Prefer `OT_ERROR_INVALID_COMMAND` slightly because we are searching for a sub command here.